src/Microsoft.ML.TimeSeries/SequentialAnomalyDetectionTransformBase.cs

```diff
@@ -16,678 +16,676 @@
-    // possible due to our design restriction. This must be fixed in the next version and will potentially affect the children classes.
-
     /// <summary>
-    /// The base class for sequential anomaly detection transforms that supports the p-value as well as the martingales scores computation from the sequence of
-    /// raw anomaly scores whose calculation is specified by the children classes. This class also provides mechanism for the threshold-based alerting on
-    /// the raw anomaly score, the p-value score or the martingale score. Currently, this class supports Power and Mixture martingales.
-    /// For more details, please refer to http://arxiv.org/pdf/1204.3251.pdf
+    /// The type of the martingale.
     /// </summary>
-    /// <typeparam name="TInput">The type of the input sequence</typeparam>
-    /// <typeparam name="TState">The type of the state object for sequential anomaly detection. Must be a class inherited from AnomalyDetectionStateBase</typeparam>
-    public abstract class SequentialAnomalyDetectionTransformBase<TInput, TState> : SequentialTransformerBase<TInput, VBuffer<Double>, TState>
-        where TState : SequentialAnomalyDetectionTransformBase<TInput, TState>.AnomalyDetectionStateBase, new()
+    public enum MartingaleType : byte
     {
         /// <summary>
-        /// The type of the martingale.
+        /// (None) No martingale is used.
         /// </summary>
-        public enum MartingaleType : byte
-        {
-            /// <summary>
-            /// (None) No martingale is used.
-            /// </summary>
-            None,
-            /// <summary>
-            /// (Power) The Power martingale is used.
-            /// </summary>
-            Power,
-            /// <summary>
-            /// (Mixture) The Mixture martingale is used.
-            /// </summary>
-            Mixture
-        }
-
+        None,
         /// <summary>
-        /// The side of anomaly detection.
+        /// (Power) The Power martingale is used.
         /// </summary>
-        public enum AnomalySide : byte
-        {
-            /// <summary>
-            /// (Positive) Only positive anomalies are detected.
-            /// </summary>
-            Positive,
-            /// <summary>
-            /// (Negative) Only negative anomalies are detected.
-            /// </summary>
-            Negative,
-            /// <summary>
-            /// (TwoSided) Both positive and negative anomalies are detected.
-            /// </summary>
-            TwoSided
-        }
+        Power,
+        /// <summary>
+        /// (Mixture) The Mixture martingale is used.
+        /// </summary>
+        Mixture
+    }
 
+    /// <summary>
+    /// The side of anomaly detection.
+    /// </summary>
+    public enum AnomalySide : byte
+    {
         /// <summary>
-        /// The score that should be thresholded to generate alerts.
+        /// (Positive) Only positive anomalies are detected.
```

Comment:
What are positive anomalies?

Reply:
Refer to my previous comment regarding documentation.

---
In reply to: [254484616](https://github.com/dotnet/machinelearning/pull/2344#discussion_r254484616) [](ancestors = 254484616)